File: antilavado-web/src/main/webapp/modules/matrix/services/matrix.service.js
/**
 * matrix.service.js
 *
 * PLAYFT.
 *
 * Copyright (c) 2015 ****Fernando Valdes <fervaldes11@gmail.com>****
 *
 * Created by ktufernando on 01/12/2015.
 */

'use strict';

//Articles service used for communicating with the articles REST endpoints
angular.module(ApplicationConfiguration.applicationModuleName)

    .factory('MatrixService', function ($resource, BASE_CONFIG) {
        return $resource(BASE_CONFIG.basePath + 'user/matrix/:clientId', {}, {
            matrixBy: {method: 'GET', params: {clientId: '@clientId'}},
            matrixForAll: {method: 'POST', params: {clientId: 'all'}},
            matrixForClient: {method: 'PUT', params: {clientId: '@clientId'}}
        })
    })

;

File: antilavado-service/src/main/java/ar/com/jf/antilavado/service/matrix/MatrixService.java
package ar.com.jf.antilavado.service.matrix;

import ar.com.jf.antilavado.repository.dto.response.matrix.RiskMatrix;
import ar.com.jf.antilavado.repository.model.MatrixMaster;

import java.util.List;

/**
 * MatrixService.java
 *
 * PLAYFT.
 *
 * Copyright (c) 2015 ****Fernando Valdes <fervaldes11@gmail.com>****
 *
 * Created by fvaldes on 01/12/2015.
 */
public interface MatrixService {

    RiskMatrix getMatrixBy(Long clientId);
    void matrixForClient(Long clientId, String username);
    void matrixForAllClients(String username);
    void matrixForAllClients();
    void save(MatrixMaster master);
}
